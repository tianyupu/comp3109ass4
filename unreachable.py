# takes in arguement of CFG
def reachable_nodes(graph):
  reachable = set()
  work = set()
  work.add(graph.root)
  while len(work) > 0:
    reachable = reachable.union(work)
    for block in work:
      work = work.union(block.out_edges)-reachable
  return reachable

