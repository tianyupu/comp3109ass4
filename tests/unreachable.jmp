  i = 1;
  s = 0;
L1:
  i = 10;
  goto L9;
L2:
  return s;
L3:
  i = 2;
  goto L8;
L4:
  return s;
L5:
  return s;
L6:
  return s;
L7:
  goto L9;
L8:
  goto L9;
L9:
  return i;

