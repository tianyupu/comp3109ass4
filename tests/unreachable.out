  i = 1;
  s = 0;
L1:
  i = 10;
  return i;
